infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java

```diff
@@ -65,7 +70,14 @@ public ParallelTransferProcessor(StateTransferBatchProcessor stateTransferBatchP
                             }
                         });
 
-        allFutures = CFUtils.allOfOrTerminateExceptionally(allFutures, batchTransferResult);
+        CompletableFuture<Void> future = MicroMeterUtils
+                .timeWhenCompletes(
+                        batchTransferResult,
+                        sample,
+                        "state-transfer.timer", "type", "committed"
+                );
+
+        allFutures = CFUtils.allOfOrTerminateExceptionally(allFutures, future);
```

Comment:
It seems to me this method is the same as CompletableFuture.allOf(...)

Reply:
allOf finishes when all the futures finish, here we need to quit once one of the futures terminate exceptionally.